lips/lip-9.md

```diff
@@ -0,0 +1,93 @@
+---
+lip: 9
+title: Libra Private Transaction Metadata
+authors: Daniel Prinz
+status: Draft
+type: Informational
+created: 09/10/2020
+---
+# Summary
+This LIP describes the conceptual model and implementation of a private, on-chain description attribute, that could be added to every P2P transaction. An example of that would be short and descriptive information about the transaction, like what are the purchased items that are being paid in a transaction. 
+
+# Abstract / Overview
+The Libra blockchain can carry informational metadata with every transaction made. This data could vary and usually used for validating both sides of the transaction could "understand" its mean. While this metadata deals with the functioning of the sender and receiver VASPs, and/or with the FATF travel rule regulatory constraints, the transaction info itself remains opaque to the end-users. This property is missing from the current payment experience of any traditional network.
+This LIP's goal is to propose a standard way of describing any transaction while keeping the following principles:
+1. Indisputable data
+2. Privacy of the end-user (sender)
+3. Ownership of the data by the end-user 
+
+
+# Terminology
+* *End users/Customers* are custodial or non-custodial wallet users which wish to transfer funds from their account to the beneficiary account (Receiver/Merchant)
+* *End-user wallet* is an on-chain VASP who provides custodian services to the end-users. This entity is considered optional and this role could be played by the end-users in a non-custodial wallet scenarios
+* *Receiver/Merchant VASP* is an entity that provides the same VASP services to the other side of the transaction, such as a regular receiver or merchant
+* *Receiver/Merchant* is the beneficiary of the transaction
+* *Private/Public keys* are keys used to sign and verify as well as encrypt and decrypt data by both sides. a private key is usually used to sign and encrypt while a public key is usually used for verification  
```

Comment:
One higher-level principle that I think is worth aligning on: what is the bar for adding a new kind of cryptographic key in Libra?

Each new key introduces a lot of complexity: someone needs securely store the private key, manage it (e.g., rotate regularly), publicize the public key, and (in this case) define a secure protocol for migrating keys from one vendor to another (if the user decides to switch). Another factor that should be taken into account is how many of these keys there will be--obviously, more keys = more complexity.

Right now (outside of validators), Libra only has account keys (one/account) and compliance keys (one/VASP). The motivation for each of these keys is quite clear--they buy a lot of use-cases. And the number of total keys is quite small--in the early days of Libra, there will probably be 10s-100s of VASPs and 100s-100000s of total accounts.

By contrast, the number of end-uses (i.e., wallet customers) will always be *much* higher than the number of VASPs or blockchain accounts. Thus, it feels like the bar for adding a new kind of cryptographic key that is (one/end-user) should be astronomically high. I don't think the utility provided by this LIP is sufficient motivation for adding this much complexity to Libra.